internal/response_decoders/common.go

```diff
@@ -77,7 +78,9 @@ func decodeArray[T any](decoder *field_decoder.FieldDecoder, decodeFunc func(*fi
 		if lengthValue.Value == -1 {
 			return nil, nil
 		}
-		return nil, decoder.WrapError(fmt.Errorf("Expected array length to be -1 or a non-negative number, got %d", lengthValue.Value))
+		// Show that length is wrong
+		decoder.PushPathContext("Length")
```

Comment:
Hmm, this is a bit weird - we did `ReadInt32Field("Length")` above, and then have to re-push `Length` to the context just to ensure the error appears at the correct place. Let's find a better way to handle this.